app/screens/buyers/BuyingItemList.java

```diff
@@ -1,44 +1,100 @@
+import java.util.List;
 import java.util.Scanner;
 
 import icebase.app.App;
 import icebase.app.MenuTitle;
+import icebase.app.api.API;
+import icebase.app.enums.CATEGORY_ENUM;
+import icebase.app.exceptions.BuyException;
+import icebase.app.exceptions.CategoryDoesNotExistException;
+import icebase.app.exceptions.UnauthorizedException;
 import icebase.app.screens.Screen;
-import java.util.Scanner;
+import icebase.app.types.Store;
+import icebase.app.types.items.Item;
+import icebase.icebase.Auth;
+import icebase.icebase.User;
 
 public class BuyingItemList implements Screen {
 
-    public void display() {
-        Scanner sc = App.sc;
-        int choice;
-        String storeName = "STORE X"; // stand-in value, should be set from constructor
-        String categoryName = "CATEGORY X"; // stand-in value, should be set from constructor
-        String[] itemList = { "Item A", "Item B", "Item C", "Return" }; // fetch from icebase
-        int range = itemList.length;
+    private String storeName;
+    private String categoryName;
+    private List<Item> itemList;
+    private ArrayList<String> itemNames;
+    private Scanner sc = App.sc;
+    private Auth auth = Auth.getAuth();
+    private User user = auth.getUser();
 
+    public BuyingItemList(Store store, CATEGORY_ENUM category) {
+        this.storeName = store.getName();
+        this.categoryName = category.value;
+        try {
+            this.itemList = API.getStoreItems(store, category);
+            for (Item item : itemList) {
+                itemNames.add(item.getName());
+            }
+            itemNames.add("return");
+
+        } catch (CategoryDoesNotExistException cdne) {
+            System.out.println(cdne.getMessage());
+        } catch (IOException io) {
+            io.printStackTrace();
+        }
+
+    }
+
+    public void display() {
         while (true) {
+            int choice;
             MenuTitle.displayStoreName(storeName);
-            MenuTitle.displaySubTitle("[" + categoryName + "]");
-            // MenuTitle.displayOptions(itemList);
+            MenuTitle.displaySubTitle(categoryName);
+            MenuTitle.displayOptions(itemNames);
 
             System.out.print("\nChoice: ");
             try {
                 choice = Integer.parseInt(sc.nextLine());
-                if (choice == range + 1) {
-                    break;
+                if (choice == itemList.size()) {
+                    return;
                 }
-                String item = itemList[choice - 1]; // fetch from icebase
-                buyItem(item);
+                Item item = itemList.get(choice - 1);
+                this.buyItem(item);
+            } catch (UnauthorizedException u) {
+                System.out.println(u.getMessage());
+            } catch (CategoryDoesNotExistException cdne) {
+                System.out.println(cdne.getMessage());
+            } catch (BuyException b) {
+                System.out.println(b.getMessage());
```

Comment:
essentially same lang gina do sa each code block, simplify by doing

```
catch (UnauthorizedException | CategoryDoesNotExistException | BuyException e) {
     System.out.println(e.getMessage());
}
```

ang reason nga want natin  icatch ang mga custom exceptions natin kay ito sila nga mga errors kay inexpect natin, `IOException` and general `Exception` lang ang mga special cases